src/core/dev/hw_queue_rx.cpp

```diff
@@ -551,13 +668,35 @@ bool hw_queue_rx::prepare_rq(uint32_t cqn)
     }
 
     m_doca_rxq.reset(rxq);
+    m_doca_ctx_rxq = doca_eth_rxq_as_doca_ctx(rxq);
+
+    err = doca_ctx_set_user_data(m_doca_ctx_rxq, {.ptr = this});
+    if (DOCA_IS_ERROR(err)) {
+        PRINT_DOCA_ERR(hwqrx_logerr, err, "doca_ctx_set_user_data ctx/hw_queue_rx: %p,%p",
+                       m_doca_ctx_rxq, this);
+        return false;
+    }
 
     err = doca_eth_rxq_set_type(m_doca_rxq.get(), DOCA_ETH_RXQ_TYPE_REGULAR);
     if (DOCA_IS_ERROR(err)) {
         PRINT_DOCA_ERR(hwqrx_logerr, err, "doca_eth_rxq_get_type_supported");
         return false;
     }
 
+    err = doca_eth_rxq_task_recv_set_conf(m_doca_rxq.get(), task_completion_cb, rx_task_error_cb,
+                                          safe_mce_sys().cq_poll_batch_max);
```

Comment:
why do we use here "cq_poll_batch_max"?
if I understand it correctly - we should be able to post big number of tasks to fill the rq, but cq_poll_batch_max refers to how many wqes to poll we do for each polling.

Reply:
This is just temporary. It will be changed.